### DISCUSSION TITLE
Can we use a reactjs component in reactpy?

### DISCUSSION THREAD

**Discussion by saurabhwadekar** [idom-team/idom#983] [Question] [answered]
Can we use react component in reactpy?

**Comment by rmorshea** [comment]
Can you provide more detail one what exactly you mean and how you would use this?

**Reply by saurabhwadekar** [reply]
I mean to say that the components created for reactjs can be wrapped in some way and used in reactpy like pynecone framework which will make development very easy for us.

**Reply by rmorshea** [reply, marked as answer]
Ah, yes. [Here are the docs](https://reactpy.dev/docs/guides/escape-hatches/javascript-components.html#custom-javascript-components). We have a [template repo](https://github.com/reactive-python/reactpy-js-component-template) as well as [some associated docs](https://reactpy.dev/docs/guides/escape-hatches/distributing-javascript.html). And lastly, here's a [real world example](https://github.com/reactive-python/reactpy-router).

**Reply by saurabhwadekar** [reply]
Thanks bro

**Reply by rmorshea** [reply]
PyneCone seems to rely on a React wrapper that dynamically installs dependencies (again this is from what I can tell). ReactPy actually used to do this, but we've found it to be somewhat unreliable. We're moving in the direction of supplying [a build tool](https://github.com/reactive-python/reactpy/issues/786) for this instead. For now, the recommended way is to use that template repository.
